src/option-group.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const mixinSetters = require('./mixin-setters')
+const setValues = require('./set-values')
+
+class OptionGroup {
+
+  constructor (text) {
+    let values = {}
+    if (typeof text === 'object') {
+      values = text
+    } else if (typeof text === 'string') {
+      values.text = text
+    }
+
+    this._attachment = null
+    this.data = {}
+
+    setValues(this, values)
+    // Adds scope of `this` to each setter fn for chaining `.get()`
+    this.addSetterScopes()
+  }
+
+  attachment (attachment) {
+    this._attachment = attachment
+
+    return this
+  }
+
+  end () {
+    return this._attachment
+  }
+
+  json () {
+    return Object.assign({}, this.data)
+  }
+
+  toJSON () {
+    return this.json()
+  }
+}
+
+// props for Slack API - true gets a generic setter fn
+const PROPS = {
+  text: true,
+  option_groups: true,
```

Comment:
I don't think that there's an `option_group` property inside an option group`, can remove this one.